org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/InMemoryJavaCompiler.java

```diff
@@ -223,7 +224,8 @@ public InMemoryJavaCompiler(ClassLoader parent, JavaVersion javaVersion) {
 	private final static MethodHandle INLINE_JSR_BYTECODE = findInlineJsrBytecode();
 	private static MethodHandle findInlineJsrBytecode() {
 		try {
-			if (JavaCore.getPlugin().getBundle().getVersion().compareTo(new Version(3, 39, 100)) >= 0) {
+			Plugin plugin = JavaCore.getPlugin();
+			if (plugin != null && plugin.getBundle().getVersion().compareTo(new Version(3, 39, 100)) >= 0) {
 				return null;
 			} else {
```

Comment:
I'd think this can all be removed. If the setter isn't present, it'll throw an exception and return null anyways

Reply:
the question is: how can we find places for removal
do you have an idea?

Reply:
Ah, I understand. You are thinking `getVersion` is a good indicator to look for obsolete code after bumping the minimal TP...

Reply:
yes

Reply:
By the way, I added a checkbox in https://github.com/eclipse/xtext/issues/2115 for keeping track of these "hacks" when we increase the minimal version of JDT core.